### DISCUSSION TITLE
Lack of theme/style animations?

### DISCUSSION THREAD

**Discussion by Spartan322** [godotengine/godot-proposals#5660] [GUI]
Its one of the things I've wondered on in regards to the GUI theming system in general is that there's no seeming capacity to create themes with animations, whether they be "continuous" or transitionary. Honestly I'm not entirely certain if its wanted or necessary in Godot's theming system in the first place, (and beyond very light messing around I haven't much experimented regarding this topic) especially since you can technically achieve mostly similar things if you choose to completely ignore the theming system to the animate the items. That's my concern however with this case, that it does raise the bar of accessibility to animating themes since you have to implement them manually anyway and I don't recall seeing a clean reusable implementation of such behavior that doesn't either do that or use multiple nodes at once. There are cases for things like style boxes specifically that continuous animations could be done with animated textures and style box texture resources, and anywhere else textures are valid you can probably get away animated textures however I'm not sure this is a great solution if this is considered a problem since you may not want to rely on textures for your style and messing with them to animate a style is not very clean in the inspector. It also doesn't solve the transitionary side of the issue anyhow.

If this considered a valuable concept by anyone, I would like further comments to discuss on the design or perhaps specific problems this might better address for which the design can or should take into account.

**Comment by Zireael07** [comment]
You should be able to animate theme/style variables with AnimationPlayer, though?

**Reply by Spartan322** [reply]
Not exactly, you can animate the state of static themes with the animation player, but as far as I'm aware that doesn't take into account UI transitions (without manually managing that yourself anyway) and unless its local to scene, it will impact more then one element of the UI. (I also don't recall animation player being capable to handle reacting to the many behaviors of Controls, though I haven't long messed with them together) And all of that doesn't rely on the theme system itself to provide, so if you say want all your buttons (or specific buttons of a specific theme) for example to do that you need to assign it to all those buttons with an animation player. There's obviously solutions that do exist to any number of the issues present on this topic, but most of them are clunky or a bit cumbersome so I just wonder if perhaps there aren't better ways it could be provided.

**Reply by Shadowblitz16** [reply]
> You should be able to animate theme/style variables with AnimationPlayer, though?

Animation player animates nodes not styles.
I would like to see a way to add animations to control themes.

Idk how it would look in code though.
